service/history/workflowExecutionUtil.go

```diff
@@ -222,3 +223,26 @@ func terminateWorkflow(
 	)
 	return err
 }
+
+func getWorkflowExecutionTimeout(namespace string, requestedTimeout int32, serviceConfig *Config) int32 {
```

Comment:
These look like duplicated code with overrideStartWorkflowExecutionRequest.

Reply:
I refactored `overrideStartWorkflowExecutionRequest` logic to also use these helpers.